### PR TITLE
Add fallback to last_active_ts if it beats the last sync time on a presence timeout.

### PR DESCRIPTION
**Pull request by Half-Shot**

### CODE REVIEW

File: synapse/handlers/presence.py [hunk 1]
```diff
@@ -1157,7 +1157,8 @@ def handle_timeout(state, is_mine, syncing_user_ids, now):
         # If there are have been no sync for a while (and none ongoing),
         # set presence to offline
         if user_id not in syncing_user_ids:
-            if now - state.last_user_sync_ts > SYNC_ONLINE_TIMEOUT:
```
Comment (erikjohnston) [inline synapse/handlers/presence.py]: Can you comment why we take the max please? kthx

**Comment by erikjohnston** [comment]
@matrixbot test this please

File: synapse/handlers/presence.py (hunk 1 above)
Comment (Half-Shot) [inline synapse/handlers/presence.py]: Done.

**Comment by erikjohnston** [comment]
@matrixbot ok to test?

**Comment by Half-Shot** [comment]
okay to merge?

maxresdefault

**Pull request merged**